obj_factory.h

```diff
@@ -102,6 +102,42 @@ class ObjFactory {
         return vec[idx.val].get();
     }
 
+    bool if_exist(const std::string& uri) {
+        if (map.find(uri) == map.end()) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    bool erase(const Idx<ObjType>& idx) {
+        if (idx.val >= vec.size()) {
+            return false;
```

Comment:
any of these 4 ? Inside the test ?
Because it is already tested
https://github.com/CanalTP/utils/pull/76/files#diff-abe94aa6106a367b2c926b2fd2c843adR167

Reply:
I was referring to real-use: if it happens during kraken life, we want to know.
Although maybe not for that one (but the next map.erase we want to know if it doesn't work, as it means the struct is corrupted)

Reply:
To be handled by the caller.